marimo/_cli/cli.py

```diff
@@ -109,6 +110,29 @@ def check_app_correctness(filename: str) -> None:
         )
 
 
+def check_app_correctness_or_convert(filename: str) -> None:
+    from marimo._convert.converters import MarimoConvert
+
+    code = ""
+    try:
+        return check_app_correctness(filename, noninteractive=True)
+    except SyntaxError:
+        # Create a single cell with the broken code
+        with open(filename, encoding="utf-8") as f:
+            code = MarimoConvert.from_text(source=f.read()).to_py()
+        with open(filename, "w", encoding="utf-8") as f:
+            f.write(code)
+    except click.ClickException:
+        # A click exception is raised if a python script could not be converted
+        with open(filename, encoding="utf-8") as f:
+            code = MarimoConvert.from_non_marimo_python_script(
```

Comment:
should `from_non_marimo_python_script` and `from_text` be inverted?

Reply:
Syntax errors must be opened as text, while a click error just means it's not a marimo notebook. I reordered the errors to highlight severity though